polaris-core/src/main/java/org/apache/polaris/core/persistence/pagination/PageTokenUtil.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.polaris.core.persistence.pagination;
+
+import static com.google.api.client.util.Preconditions.checkArgument;
+
+import jakarta.annotation.Nullable;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Optional;
+
+final class PageTokenUtil {
+
+  private PageTokenUtil() {}
+
+  static PageToken decodePageRequest(
+      @Nullable String requestedPageToken, @Nullable Integer requestedPageSize) {
+    int pageSize;
+    String encodedDataReference = null;
+    if (requestedPageToken != null) {
```

Comment:
Looks like you've changed the proposed format of the tokens here

Reply:
It is; closing as duplicate of the other threads asking how this format can be extended